heat-eq-simulation/1D_ftcs.py

```diff
@@ -0,0 +1,53 @@
+# Forward Time Centered Space (ftcs) is an explicit numerical method
+# for derving the solution to PDEs
+
+# we will use the ftcs method below to simulate the 1D Heat Equation
+import numpy
+import matplotlib.pyplot as pyPlot
+
+# Global Constants
+
+lengthOfRod = 10
+maxTime = 1
+heatConductivity = 1 #100 is going the least in nagtive and 1000 produces the same graph to my eye
+numPointsSpace = 200
+numPointsTime = 2000
+
+# Length Vector plotted on x-axis
+xSpace = numpy.linspace(0, lengthOfRod, numPointsSpace)
+# Time Vector plotted on y-axis
+timeSpace = numpy.linspace(0, maxTime, numPointsTime)
+
+timeStepSize = timeSpace[1] - timeSpace[0]
+spaceStepSize = xSpace[1] - xSpace[0]
+
+boundaryConditions = [numpy.sin(lengthOfRod)+numpy.sin(maxTime), 
```

Comment:
So, if I understand this well, your `boundaryConditions` should a list of two numpy arrays: the boundary conditions on the left and on the right. However, the way you've written this, `boundaryConditions` is a list of two **numbers**. 

Perhaps you meant something like 
```python
numpy.sin(lengthOfRod)+numpy.sin(timeSpace)
```

In any case, the boundary and initial conditions shoud be compatible. The following conditions should be satisfied:

- $\alpha(0) = u^0(0)$
- $\beta(0) = u^0(L)$, where $L$ is the length of the beam

It would be good to even put an assertion in your code to make sure these are satisfied.

Also, you are free to choose any initial conditions you like. You can choose some simpler conditions for starters.

Reply:
See also #4

Reply:
wrapped it in a numpy.array, does that address the issue?

Reply:
@debdal6 following our discussion this week, I believe the issue should be fixed. Can you please update the code? It would be good to have a couple of `assert`s to make sure the initial/boundary conditions are compatible (check the notes I've shared with you).